File: Backend/Owl.Overdrive.Business/DTOs/GameDtos/Display/Details/GameDetailsDto.cs
using Owl.Overdrive.Domain.Enums;

namespace Owl.Overdrive.Business.DTOs.GameDtos.Display.Details
{
    public class GameDetailsDto
    {
        public long Id { get; set; }
        public string Name { get; set; } = null!;
        public string? Story { get; set; }
        public DateTime? FirstReleaseDate { get; set; }
        public string? Publisher { get; set; }
        public GameDetailsGeneral Details { get; set; } = new();
        public List<GameDetailsMultiplayerModeDto> MultiplayerModes { get; set; } = new();
        public GameDetailsSpellingsDto Spellings { get; set; } = new();
        public List<GameDetailsWebsiteDto> Websites { get; set; } = new();
        public GameDetailsSupportedLanguageDto Supportedlanguages { get; set; } = new();
        public List<GameDetailsPlatformReleasedDatesDto> PlatformsReleaseDates { get; set; } = new();
        public GameCoverDetailsDto? Cover { get; set; }
        public GameBackgroundDetailsDto? Background { get; set; }
    }

    public class GameDetailsPlatformReleasedDatesDto
    {
        public string PlatformName { get; set; } = null!;
        public List<GameDetailsReleasedDateDto> ReleaseDates { get; set; } = new();
    }

    public class GameDetailsReleasedDateDto
    {
        public DateTime? Date { get; set; }
        public long? RegionId { get; set; }
        public EGameStatus? Status { get; set; }
    }

    
}

File: Frontend/src/app/data/types/game/game-details-dto.ts
import { EWebsites } from "src/app/core/enums/enum-websites"

export interface GameDetailsDto {
    id: number
    name: string 
    firstReleaseDate: Date 
    story: string 
    details: GameDetailsGeneral
    multiplayerModes: Array<GameDetailsMultiplayerModeDto>
    spellings: GameDetailsSpellingsDto
    websites: Array<GameDetailsWebsiteDto>
    supportedlanguages:GameDetailsSupportedLanguageDto
    platformsReleaseDates: Array<GameDetailsPlatformReleasedDatesDto>
    cover: GameCoverDetailsDto
    background: GameBackgroundDetailsDto
}

export interface GameCoverDetailsDto {
    imageTitle: string 
    imageData : string
}

export interface GameBackgroundDetailsDto {
    imageTitle: string 
    imageData : string
}

export interface GameDetailsGeneral
{
    description: string
    platformsDtos: Array<GameDetailsGeneralPlatformsDto>
    companies: GameDetailsGeneralInvolvedCompanies
    themes: Array<string>
    genres: Array<string>
    gameModes: Array<string>
    playerPerspectives: Array<string>
}

export interface GameDetailsGeneralPlatformsDto
{
    id: number
    name: string
}

export interface GameDetailsGeneralInvolvedCompanies
{
    developer: Array<GameDetailsGeneralCompany>
    publisher: Array<GameDetailsGeneralCompany>
    porting: Array<GameDetailsGeneralCompany>
    supporting: Array<GameDetailsGeneralCompany>
}

export interface GameDetailsGeneralCompany
{
    id: number
    name: string
}

export interface GameDetailsMultiplayerModeDto {
    platformName: string
    campaignCoop: boolean
    dropIn: boolean
    lanCoop: boolean
    offlineCoop: boolean
    offlineCoopMax: number
    offlineMax : number
    onlineCoop: boolean
    onlineCoopMax: number
    onlineMax : number
    splitScreen: boolean
    splitScreenOnline : boolean
    platformId : number
}

export interface GameDetailsSpellingsDto
{
    localizedTitles: Array<GameDetailsTitlesDto>
    alternativeTitles: Array<GameDetailsTitlesDto>
}

export interface GameDetailsTitlesDto
{
    type: string
    title: string
}

export interface GameDetailsWebsiteDto {
    url: string
    category: EWebsites
}

export interface GameDetailsSupportedLanguageDto
{
    audio:Array<number>
    subtitles:Array<number>
    interface:Array<number>
}

export interface GameDetailsPlatformReleasedDatesDto
{
    platformName: string
    releaseDates: Array<GameDetailsReleasedDateDto>
}

export interface GameDetailsReleasedDateDto
{
    date: Date
    regionId: number
    status: number
}